Model/src/GamePackage/Player.java

```diff
@@ -1,4 +1,265 @@
+		return wallet;
+	}
+
+	/******************************************************************
+	 * Setter method for wallet variable.
+	 * 
+	 * @param wallet Sets the int wallet.
+	 *****************************************************************/
+	public void setWallet(int wallet) {
+		this.wallet = wallet;
+	}
+
+	/******************************************************************
+	 * Getter method for propertiesOwned ArrayList.
+	 * 
+	 * @return propertiesOwned Returns the ArrayList propertiesOwned.
+	 *****************************************************************/
+	public ArrayList<BoardSquare> getPropertiesOwned() {
+		return propertiesOwned;
+	}
+
+	/******************************************************************
+	 * Setter method for propertiesOwned ArrayList.
+	 * 
+	 * @param propertiesOwned Sets the ArrayList propertiesOwned.
+	 *****************************************************************/
+	public void setPropertiesOwned(ArrayList<BoardSquare> 
+	propertiesOwned) {
+		this.propertiesOwned = propertiesOwned;
+	}
+
+	/******************************************************************
+	 * Getter method for cardsHeld ArrayList.
+	 * 
+	 * @return cardsHeld Returns the ArrayList cardsHeld.
+	 *****************************************************************/
+	public ArrayList<Card> getCardsHeld() {
+		return cardsHeld;
+	}
+
+	/******************************************************************
+	 * Setter method for cardsHeld ArrayList.
+	 * 
+	 * @param cardsHeld Sets the ArrayList cardsHeld.
+	 *****************************************************************/
+	public void setCardsHeld(ArrayList<Card> cardsHeld) {
+		this.cardsHeld = cardsHeld;
+	}
+	
+	/******************************************************************
+	 * Checks to make sure player has sufficient funds before
+	 * subtracting amount from wallet. If the player has sufficient
+	 * funds the method subtracts amount from wallet and returns the
+	 * amount paid. If player doesn't have sufficient funds he method
+	 * subtracts what is able to be paid from wallet and returns the
+	 * amount that has been paid.
+	 * 
+	 * @param amount The amount of money requested from player.
+	 * @return amountPaid The amount of money that was paid by player.
+	 *****************************************************************/
+	public int pay(int amount) {
+		int amountPaid = 0;
+		
+		if(wallet >= amount) {
+			wallet -= amount;
+			amountPaid = amount;
+		}
+		else {
+			amountPaid = amount - wallet;
+			wallet -= amountPaid;
+		}
+		
+		return amountPaid;
+	}
+	
+	/******************************************************************
+	 * Removes the property from the player's ArrayList of properties
+	 * owned.
+	 * 
+	 * @param property The property to be removed from owned properties
+	 * list.
+	 *****************************************************************/
+	public void sell(BoardSquare property) {
+		propertiesOwned.remove(property);
+	}
+	
```

Comment:
I think we can remove the sell() function. I am having a hard time remembering why we had this here. The original issue has:
- giveProperty(): which should removes a property. The idea here is we call Player.giveProperty() so its removed 
- recieveProperty(): which should add a property. The idea here is we call Player.receiveProperty() so its added. 

I think it makes more sense for the Game class to have a sell method that makes calls to this. We should all have a discussion about this though.

Reply:
the sell method was the way the bank collected payments from players.

Reply:
I forget, why cant we just use pay()? Cant Game say x player needs to pay bank, call x.pay(amount) and give amount to bank?

Reply:
remove sell method